torchvision/models/detection/transform.py

```diff
@@ -65,14 +73,16 @@ class GeneralizedRCNNTransform(nn.Module):
     It returns a ImageList for the inputs, and a List[Dict[Tensor]] for the targets
     """
 
-    def __init__(self, min_size, max_size, image_mean, image_std):
+    def __init__(self, min_size, max_size, image_mean, image_std, size_divisible=32, fixed_size=None):
```

Comment:
I'm ok with keeping the `fixed_size` as is for now to unblock, but we definitely will need to initiate a work on properly doing abstractions for joint transforms

Reply:
Yes this is a weird workaround. As discussed offline in the past, the `GeneralizedRCNNTransform` needs to be abstracted and SSD should have its own similar class.

Reply:
If it is only in the inference stage, the `GeneralizedRCNNTransform` is pretty friendly.

But if we want to train a model, the current embedding way will cause some redundancy with the other data augmentations methods before `GeneralizedRCNNTransform`.

Reply:
The challenge is that `GeneralizedRCNNTransform` does too many things and fixed input detectors like SSD don't need these. Perhaps we can have a similar very friendly transform for those methods that don't resize using the min/max_size logic, instead of putting even more functionality in this class... We definitely need to think about this. :)

Reply:
The way I see this in the future is that we will have different `transform` for different models
```python
SSDTransform(...)
GeneralizedRCNNTransform(...)
DETRTranform(...)
```
and the way to avoid too much code duplication will be by having nice abstractions for the joint transforms, so that each one of those will be able to be easily implemented. Something like
```python
GeneralizedRCNNTransform = Compose(ResizeWithMinSize(...), RandomFlip(...), Normalize(...),)
```
But we are not there yet